hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/ReadableNodeStoreImpl.java

```diff
@@ -77,4 +85,21 @@ protected <T extends ReadableKVState<EntityNumber, Node>> T nodesState() {
     public Iterator<EntityNumber> keys() {
         return nodesState().keys();
     }
+
+    private Roster constructFromNodesState(@NonNull final ReadableKVState<EntityNumber, Node> nodesState) {
+        final var rosterEntries = new ArrayList<RosterEntry>();
+        for (final Iterator<EntityNumber> it = nodesState.keys(); it.hasNext(); ) {
+            final var nodeNumber = it.next();
+            final var nodeDetail = nodesState.get(nodeNumber);
+            final var entry = RosterEntry.newBuilder()
```

Comment:
Do we need a test for `if(!nodeDetail.isDeleted())` here so we don't include deleted nodes in the roster?

Reply:
Makes sense to me. I'll ask to be sure

Reply:
Yes we want to check for deleted nodes, and not include them. Fixed.